### PR TITLE
feat: add filter component

### PR DESCRIPTION
**Pull request by kboutsika**
## Description

implement the a simple filter component based on the design system (no search, no multi selection)

## Screenshot

Στιγμιότυπο 2021-05-14, 2 59 25 μμ
Στιγμιότυπο 2021-05-14, 2 59 17 μμ

### CODE REVIEW

File: src/components/Filter/utils.ts [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+  buttonType,
+  styleType,
+}: BackgroundColorProps) => {
+  if (open) {
+    return defineBackgroundColor(theme, activeCalculatedColor);
+  } else if (hasSelectedValue || styleType === 'transparent') {
+    return 'transparent';
+  } else if (styleType === 'filled' || styleType === 'outlined') {
+    return defineBackgroundColor(theme, calculatedColor, buttonType);
+  } else {
```
Comment (panvourtsis) [inline src/components/Filter/utils.ts:22]: you don't need an `else` here as a simple return would do the trick

File: src/components/Filter/Filter.stories.mdx [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+# Filter
+
+A universal Filter component that is a clickable element that can hold many forms.
+
+# Usage
+
+```js
+import { Filter } from '@orfium/ictinus';
+
+<Filter onClick={() => {}}>Click Me</Filter>;
```
Comment (panvourtsis) [inline src/components/Filter/Filter.stories.mdx:17]: this usage example is not quite right based on what i saw of your example. Please use a minimal example

File: src/components/Filter/Filter.style.ts [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+`;
+
+export const buttonSpanStyle = () => () => {
+  return {
+    display: 'flex',
+    alignItems: 'center',
+  };
+};
+
+export const optionsStyle = () => (theme: Theme) => css`
```
Comment (panvourtsis) [inline src/components/Filter/Filter.style.ts:21]: this is being used on the `menu` component. As this is used in 2 places now can we make a simple one to handle both cases?

File: src/components/Filter/types.ts [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+  /** Defines the style type of the button */
+  styleType: StyleType;
+};
+
+export type TestProps = {
+  dataTestId?: string;
+};
+
+
+export type BackgroundColorProps = {
```
Comment (panvourtsis) [inline src/components/Filter/types.ts:38]: as you use often the following

```
open: boolean;
  theme: Theme;
  hasSelectedValue: boolean;
  calculatedColor: ColorShapeFromComponent;
  activeCalculatedColor: ColorShapeFromComponent;
```
you can make a base type and use this as extension

**Review by panvourtsis — CHANGES_REQUESTED** [review]

File: src/components/Filter/Filter.stories.mdx (hunk 2 above)
Comment (kboutsika) [reply src/components/Filter/Filter.stories.mdx]: yes you are right i changed it

**Comment by panvourtsis** [comment]
forgot to mention that we need one playground story for this with all the properties in knobs

**Comment by panvourtsis** [comment]
other than that this PR LGTM 🎊

**Review by panvourtsis — APPROVED** [review]

**Pull request merged**